File: src/ru/hudspeed/pro/CameraPoint.java
package ru.gpsantiradar.app;

/** One RadarBase object stored in the local database. */
public final class CameraPoint {
    public long id;
    public double latitude;
    public double longitude;
    public int type;
    /** RadarBase direction kind: 0 all, 1 oncoming, 2 both ways, 3 rear, 4 front+rear. */
    public int dirType;
    /** Direction of the controlled traffic flow, in degrees. */
    public float direction;
    public int distanceMeters;
    public int reverseDistanceMeters;
    /** Full RadarBase sector angle, in degrees. */
    public float angleDegrees;
    public float rank;
    public boolean newbie;
    /** Compact representation of RadarBase speedControls. */
    public String speedRules = "";

    public CameraPoint() {}

    public int currentSpeedLimit() {
        return SpeedControlRules.currentCarLimit(speedRules, System.currentTimeMillis());
    }

    public boolean isObservation() { return type == 15; }
    public boolean isFake() { return type == 16; }
    public boolean isAverageSpeed() { return type == 41 || type == 42 || type == 43; }
    public boolean isCameraOrControl() { return RadarBaseTypes.isCameraOrControl(type); }
    public boolean isRoadObject() { return !isCameraOrControl(); }

    public boolean hasReverseZone() {
        return reverseDistanceMeters > 0 && (dirType == 2 || dirType == 3 || dirType == 4);
    }

    public String typeName() { return RadarBaseTypes.name(type); }

    public String directionName() {
        switch (dirType) {
            case 0: return "во всех направлениях";
            case 1: return "навстречу потоку";
            case 2: return "два встречных направления";
            case 3: return "в спину потоку";
            case 4: return "в лицо и в спину потоку";
            default: return "направление не указано";
        }
    }
}

File: src/main/kotlin/ru/gpsantiradar/app/CameraPoint.kt
package ru.gpsantiradar.app

/** One RadarBase object stored in the local database. */
class CameraPoint {
    var id: Long = 0
    var latitude: Double = 0.0
    var longitude: Double = 0.0
    var type: Int = 0
    /** RadarBase direction kind: 0 all, 1 oncoming, 2 both ways, 3 rear, 4 front+rear. */
    var dirType: Int = 0
    /** Direction of the controlled traffic flow, in degrees. */
    var direction: Float = 0f
    var distanceMeters: Int = 0
    var reverseDistanceMeters: Int = 0
    /** Full RadarBase sector angle, in degrees. */
    var angleDegrees: Float = 0f
    var rank: Float = 0f
    var newbie: Boolean = false
    /** Compact representation of RadarBase speedControls. */
    var speedRules: String = ""

    fun currentSpeedLimit(): Int = SpeedControlRules.currentCarLimit(speedRules, System.currentTimeMillis())
    fun isObservation(): Boolean = type == 15
    fun isFake(): Boolean = type == 16
    fun isAverageSpeed(): Boolean = type == 41 || type == 42 || type == 43
    fun isCameraOrControl(): Boolean = RadarBaseTypes.isCameraOrControl(type)
    fun isRoadObject(): Boolean = !isCameraOrControl()
    fun hasReverseZone(): Boolean = reverseDistanceMeters > 0 && (dirType == 2 || dirType == 3 || dirType == 4)
    fun typeName(): String = RadarBaseTypes.name(type)

    fun directionName(): String = when (dirType) {
        0 -> "во всех направлениях"
        1 -> "навстречу потоку"
        2 -> "два встречных направления"
        3 -> "в спину потоку"
        4 -> "в лицо и в спину потоку"
        else -> "направление не указано"
    }
}
